### PR TITLE
fix(samples): Allow user to set the namespace and name  for the sample as parameters.

### PR DESCRIPTION
**Pull request by ScrapCodes**
…

**Description of your changes:**

A user might want to run the experiment again and again and thus choose a different name each time. This will not require him to compile and upload the pipeline each time.
Similarly, a users namespaces may be different from "anonymous".

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [x] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: samples/contrib/kubeflow-katib/mpi-job-horovod.py [hunk 1]
```diff
@@ -32,10 +32,9 @@
     name="Launch Katib MPIJob Experiment",
     description="An example to launch Katib Experiment with MPIJob"
 )
-def horovod_mnist_hpo():
-    # Experiment name and namespace.
-    experiment_name = "mpi-horovod-mnist"
-    experiment_namespace = "anonymous"
+def horovod_mnist_hpo(
+        experiment_name = "mpi-horovod-mnist",
```
Comment (Ark-kun) [inline samples/contrib/kubeflow-katib/mpi-job-horovod.py:36]: How about this formatting:
```python
def horovod_mnist_hpo(
    experiment_name: str = "mpi-horovod-mnist",
    experiment_namespace: str = "anonymous",
):
```

Comment (ScrapCodes) [reply samples/contrib/kubeflow-katib/mpi-job-horovod.py]: I agree with this, since this is followed at other places as well.

**Pull request merged**